packages/netlify-cms-widget-map/package.json

```diff
@@ -30,6 +30,6 @@
     "react-immutable-proptypes": "^2.1.0"
   },
   "dependencies": {
-    "ol": "^5.3.1"
+    "ol": "^6.9.0"
```

Comment:
I think this version has breaking changes and can't be updated as is.
It breaks the map widget, see the deploy preview:
![image](https://user-images.githubusercontent.com/26760571/144858939-3b532de6-6b10-4fef-ae93-41bbce440443.png)

https://deploy-preview-6009--cms-demo.netlify.app/#/collections/kitchenSink/entries/a-big-entry-with-all-the-things

Reply:
Any suggestions on what to to here?

The reason for creating the PR is because we have a non-negotiable strict CSP policy and there are `new Function` eval code being emitted from netlify-cms and its dependencies as a colleague of mine outline in #2138.

Our other alternative is to use a different CMS and judging by this PR, that might be the easier option.

Fixing this and the ajv schema `new Function` emissions is not something I am relishing.

I don't know the codebase at all.

Reply:
I don't see any tests in the `netlify-cms-widget-map` package, should there be?

Reply:
I don't see any tests for this widget

Reply:
Hi @dagda1, we don't have automated tests for this widget, but if you try to use the deploy preview (the link I shared), you'd see it's broken.

The breaking changes are documented in https://github.com/openlayers/openlayers/releases/tag/v6.0.0, if you'd like to take a look.

I understand the CSP issue, but we'd like to avoid breaking another feature of the CMS while we handle it

Reply:
@erezrokah no tests in the package and no types make this really difficult to someone who is looking at this for the first time.  

I was hoping for at least some collaboration with someone who actually knows this code but instead, I have been given a link.

Not a lot for me to go on here.

Is the bad CSP not important to netlify?

Reply:
@erezrokah I have [over 1000 open source contributions](https://github.com/dagda1) in the last year and this is without question the least help I received from one of the biggest companies.

Also, what I am suggesting is of benefit so I am pretty shocked to say the least.

Reply:
Hi @dagda1, sorry for creating this impression.

The CMS is a community project and thus most of the work is expected to come from external contributors.

I understand the benefit of fixing the CSP issue, and the importance of the issue can also be seen based on the upvotes [it receives from the community](https://github.com/netlify/netlify-cms/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc).

I do think that breaking one feature in order to fix another issue is not something that we should introduce.

For the lack of tests and types, this has been an ongoing effort and we've been introducing both when we make a code change to an area that doesn't have them.

The best way to move forward on this would be to look at the map's widget code here:
https://github.com/netlify/netlify-cms/blob/f1e5dbc9ea75570ffe5163cfdad3245699fca378/packages/netlify-cms-widget-map/src/withMapControl.js#L72
and understand the necessary changes required for it to support the major version bump of the `ol` library.

That's the only place we're using the `ol` library so the fix should be scoped to that widget.

Reply:
> and understand the necessary changes required for it to support the major version bump of the ol library.

I have found that code some time ago and the reason the widget is not rendering is because  the map container's width or height are 0.

[This warning](https://github.com/openlayers/openlayers/blob/c1f4749e4d4701c73be1c3787a700d0fed8cfd7d/src/ol/PluggableMap.js#L1530) is triggered

Reply:
> I do think that breaking one feature in order to fix another issue is not something that we should introduce.

who said anything snout committing broken code.  all i am asking gif and totally not getting is done input from someone who knows the code better than me.





> The best way to move forward on this would be to look at the map's widget code here:
> 
> https://github.com/netlify/netlify-cms/blob/f1e5dbc9ea75570ffe5163cfdad3245699fca378/packages/netlify-cms-widget-map/src/withMapControl.js#L72
> 

that is as i said the least helpful advice i have received in over 1000 commits to open source projects this year.

look at the code is what you are saying.

no sh*t sherlock.  some things don't need to be stayed

Reply:
@erezrokah If I give the container div a fixed height then if you ask me, the widget works.

Any idea why the container div would not have a 0 height:

![netlify-shit](https://user-images.githubusercontent.com/118328/145042424-0f75b191-0182-4be7-be48-6093a2dfe40c.png)

Reply:
Woo, great find. Let me try and isolate this problem locally.
They did a whole lot of style changes in the new version, so maybe this is just about setting a `min-height`

Reply:
> 

Thank you

Reply:
So adding `height: 400px;` after this line https://github.com/netlify/netlify-cms/blob/9f0d11957c139cc274d4ccde0ac537d37af3b9b2/packages/netlify-cms-widget-map/src/withMapControl.js#L82 seems to fix this issue.

I can't seem to push this change as it says I don't have access to the fork, do you mind doing this change?

Reply:
@erezrokah I could add 400px but would it not be better to make height a prop with 400px being the default?

Giving everyone 400px sounds less than ideal although I'm happy to do that

Reply:
My guess is that at some point it became the container responsibility to handle this, but I couldn't find the change that introduced it

Reply:
> I could add 400px but would it not be better to make height a prop with 400px being the default?

Even better!

Reply:
@erezrokah I've done that in [this commit](https://github.com/netlify/netlify-cms/pull/6009/commits/a1dba53f05b1bb466c0949ff20fcec87a23f9637) and the map is rendering.

I've also reset `devtool` to `source-map`.

Are there any other checks we need to make?

Reply:
Added another comment here https://github.com/netlify/netlify-cms/pull/6009#discussion_r764160640